detectable/src/main/java/com/synopsys/integration/detectable/detectables/bazel/BazelExtractor.java

```diff
@@ -73,17 +73,21 @@ public BazelExtractor(
         this.bazelProjectNameGenerator = bazelProjectNameGenerator;
     }
 
-    public Extraction extract(ExecutableTarget bazelExe, File workspaceDir, File workspaceFile) throws DetectableException, ExecutableFailedException {
-        toolVersionLogger.log(workspaceDir, bazelExe, "version");
-        BazelCommandExecutor bazelCommandExecutor = new BazelCommandExecutor(executableRunner, workspaceDir, bazelExe);
-        Pipelines pipelines = new Pipelines(bazelCommandExecutor, bazelVariableSubstitutor, externalIdFactory, haskellCabalLibraryJsonProtoParser);
-        Set<WorkspaceRule> workspaceRulesFromFile = parseWorkspaceRulesFromFile(workspaceFile);
-        Set<WorkspaceRule> workspaceRulesToQuery = workspaceRuleChooser.choose(workspaceRulesFromFile, workspaceRulesFromDeprecatedProperty, workspaceRulesFromProperty);
-        CodeLocation codeLocation = generateCodelocation(pipelines, workspaceRulesToQuery);
-        return buildResults(codeLocation, bazelProjectNameGenerator.generateFromBazelTarget(bazelTarget));
+    public Extraction extract(ExecutableTarget bazelExe, File workspaceDir, File workspaceFile) {
+        try {
+            toolVersionLogger.log(workspaceDir, bazelExe, "version");
+            BazelCommandExecutor bazelCommandExecutor = new BazelCommandExecutor(executableRunner, workspaceDir, bazelExe);
+            Pipelines pipelines = new Pipelines(bazelCommandExecutor, bazelVariableSubstitutor, externalIdFactory, haskellCabalLibraryJsonProtoParser);
+            Set<WorkspaceRule> workspaceRulesFromFile = parseWorkspaceRulesFromFile(workspaceFile);
+            Set<WorkspaceRule> workspaceRulesToQuery = workspaceRuleChooser.choose(workspaceRulesFromFile, workspaceRulesFromDeprecatedProperty, workspaceRulesFromProperty);
+            CodeLocation codeLocation = generateCodelocation(pipelines, workspaceRulesToQuery);
+            return buildResults(codeLocation, bazelProjectNameGenerator.generateFromBazelTarget(bazelTarget));
+        } catch (Exception e) {
+            return new Extraction.Builder().exception(e).build();
+        }
```

Comment:
We probably want to just let the exception bubble up

Reply:
Done